pkgs/build-support/zig/build-zig-package/default.nix

```diff
@@ -0,0 +1,44 @@
+{ lib, stdenv, zig }:
+argsFun:
+
+let
+  wrapDerivation = f:
+    stdenv.mkDerivation (finalAttrs:
+      f (lib.toFunction argsFun finalAttrs)
```

Comment:
When you change `f` to be an overlay, the composition here becomes a lot like `lib.extends`, except the first layer doesn't take a `prev` argument.
I think we could use a helper for this.

Reply:
> I think we could use a helper for this.

Actually just `lib.extends`.
Change `f` to take `finalAttrs: prevAttrs:` and then:

```diff
-    stdenv.mkDerivation (finalAttrs:
-      f (lib.toFunction argsFun finalAttrs)
+    stdenv.mkDerivation (lib.extends f (lib.toFunction argsFun))
```